src/Symfony/Component/DependencyInjection/Compiler/ResolveInvalidReferencesPass.php

```diff
@@ -91,6 +92,10 @@ private function processArguments(array $arguments, $inMethodCall = false)
                 if (!$exists && ContainerInterface::NULL_ON_INVALID_REFERENCE === $invalidBehavior) {
                     $arguments[$k] = null;
                 } elseif (!$exists && ContainerInterface::IGNORE_ON_INVALID_REFERENCE === $invalidBehavior) {
+                    if ($inCollection) {
+                        unset($arguments[$k]);
+                        continue;
```

Comment:
are you sure this should happen before throwing the exception just below (I didn't check)?